optimisation: Improve Stop Print menu

gudnimg opened the pull request:
Optimise the "Stop print" menu by using the menu API methods. **UI is visually unchanged.**



Before:

```
Program:  234486 bytes (89.4% Full)
(.text + .data + .bootloader)
 
Data:       5680 bytes (69.3% Full)
(.data + .bss + .noinit)
```

After:
```
Program:  234370 bytes (89.4% Full)
(.text + .data + .bootloader)
 
Data:       5680 bytes (69.3% Full)
(.data + .bss + .noinit)
```

Change in memory:
Flash: -116 bytes
SRAM: 0 bytes

gudnimg committed:
optimisation: Stop print menu

There is no need to manually track the cursor position.
We can use the menu API:
MENU_BEGIN()
...
// menu items in between here
...
MENU_END()

Change in memory:
Flash: -116 bytes
SRAM: 0 bytes

leptun commented:
I have not tested this, but I don't think this is ok. To me it looks as if the menu is blocking. The old menu would allow the rest of the firmware to work while the prompt is displayed. Is it the case with this new one or does it prevent the printer from printing while it is displayed?

gudnimg commented:
> I have not tested this, but I don't think this is ok. To me it looks as if the menu is blocking. The old menu would allow the rest of the firmware to work while the prompt is displayed. Is it the case with this new one or does it prevent the printer from printing while it is displayed?

Just double-checked and you're right. If I wait long enough, the current move will finish and then the printer halts until the menu goes away.

gudnimg commented:
@leptun I've fixed the issue you mentioned while still reducing the flash memory consumption. I removed my changes to "Resume print" menu items as they are not really relevant here.

3d-gussner reviewed (approved):
Tested on MK404 and works as expected
